utils/main_snowplow_bigquery.py

```diff
@@ -0,0 +1,249 @@
+# coding=utf-8
+# Copyright 2022-2023 Snowplow, Google LLC..
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Loads the data needed to run Fractribution into Bigquery."""
+
+import os
+import re
+from typing import Any, Dict, List, Mapping, Optional, Tuple
+from absl import app
+from absl.flags import argparse_flags
+import fractribution
+from google.cloud import bigquery
+
+# Construct a BigQuery client object with authentication from JSON
+SERVICE_ACCOUNT_JSON = os.environ["google_application_credentials"]
+client = bigquery.Client.from_service_account_json(SERVICE_ACCOUNT_JSON)
+
+project_id = os.environ.get('project_id')
+dataset = os.environ.get('bigquery_dataset')
+
+_OUTPUT_TABLES = ["snowplow_fractribution_path_summary_with_channels", "snowplow_fractribution_report_table", "snowplow_fractribution_channel_attribution"]
+
+VALID_CHANNEL_NAME_PATTERN = re.compile(r"^[a-zA-Z_]\w+$", re.ASCII)
+
+def _is_valid_column_name(column_name: str) -> bool:
+    """Returns True if the column_name is a valid Bigquery column name."""
+
+    return (
+        len(column_name) <= 300
+        and VALID_CHANNEL_NAME_PATTERN.match(column_name) is not None
+    )
+
+
+def _extract_channels(client, params: Mapping[str, Any]) -> List[str]:
+    """Returns the list of names by running extract_channels.sql.
+
+    Args:
+      client: Client.
+      params: Mapping of template parameter names to values.
+    Returns:
+      List of channel names.
+    Raises:
+      ValueError: User-formatted error if channel is not a valid Bigquery column.
+    """
```

Comment:
If I'm being picky, this function doesn't use the params: argument at all

Reply:
I got this removed for all warehouses, none uses it in the end, must have been a leftover unused code.